libraries/tentacle.rb

```diff
@@ -41,10 +43,31 @@ def installer_url(version)
       "https://download.octopusdeploy.com/octopus/Octopus.Tentacle.#{version}-x64.msi"
     end
 
+    def tentacle_exists?(server, api_key, thumbprint)
+      ::JSON.parse(api_client(server, api_key).get('/machines/all')).any? do |machine|
```

Comment:
@spuder this is what I was talking about before in regards to the build in http client